stores/facilitiesStore.ts

```diff
@@ -36,23 +36,37 @@ export const useFacilitiesStore = defineStore(
             facilityData.value = facilityResults
         }
 
-        async function updateFacility(facility: MutationUpdateFacilityArgs) {
+        async function updateFacility(facility: MutationUpdateFacilityArgs):
+        Promise<DeleteResult | undefined> {
             try {
                 return await graphQLClientRequestWithRetry(
                     gqlClient.request.bind(gqlClient),
                     updateExistingFacilityGqlMutation,
                     facility
-                )
+                ) as DeleteResult
             } catch (error) {
                 console.error('Failed to update facility:', error)
             }
         }
 
+        async function deleteFacility(facilityId: MutationDeleteFacilityArgs) {
+            try {
+                return await graphQLClientRequestWithRetry(
+                    gqlClient.request.bind(gqlClient),
+                    deleteExistingFacilityGqlMutation,
+                    facilityId
+                )
+            } catch (error) {
```

Comment:
we should return a failure state to the calling function so they can display a nice error to the user if this fails